salt/states/git.py

```diff
@@ -2238,13 +2238,18 @@ def detached(name,
 
         local_commit_id = _get_local_rev_and_branch(target, user, password)[0]
 
-        if remote_ref_type is 'hash' \
-                and __salt__['git.describe'](target,
-                                             ref,
-                                             user=user,
-                                             password=password):
-            # The ref is a hash and it exists locally so skip to checkout
-            hash_exists_locally = True
+        if remote_rev_type is 'hash':
+            try:
+                __salt__['git.describe'](target,
+                                         rev,
```

Comment:
In this branch `ref` is still used, so this variable isn't declared - did you mean to put this in a different branch?